### DISCUSSION TITLE
override client assertion for private_key_jwt

### DISCUSSION THREAD

**Discussion by sanisc** [panva/openid-client#391] [Q&A]
Hello,

My client assertion is signed by KMS so i can't use the keystore options in the library is there any way to override this and provide my own client_assertion as part of the request to the token endpoint?

**Comment by panva** [comment]
Hi @sanisc 

~No, there is not.~

**Reply by sanisc** [reply]
i managed to work around this by using "none" on the token_endpoint_auth_method so i could then just submit what i wanted in exchangebody on the client.callback.

**Reply by panva** [reply]
nice trick
